project-koku/koku
# Small cost model API fixes

Changes proposed in this PR:
* Quick fix to cost models to add currency to the open api docs
* Also a quick fix to the filtering caught to filter by currency

Testing Instructions:
Test API Docs
1. Pull branch
2. Copy file under docs/source/specs
3. insert data into [swagger-editor](https://editor.swagger.io/)
4. Ensure Cost models has currency documented as a filter

Test Filtering:
1. Create and Load test customer data
2. hit [sources](http://127.0.0.1:8000/api/cost-management/v1/cost-models/) Endpoint
3. Copy the body of an existing source and draft up a [POST]http://127.0.0.1:8000/api/cost-management/v1/cost-models/ with the existing body without the UUID
```
Example body:
 {
            "name": "Cost Management AWS Cost Model",
            "description": "A cost model for markup on AWS costs.",
            "source_type": "AWS",
            "created_timestamp": "2022-04-26T19:19:38.012720Z",
            "updated_timestamp": "2022-04-26T19:19:38.012739Z",
            "rates": [],
            "markup": {
                "value": "10.0000000000",
                "unit": "percent"
            },
            "distribution": "cpu",
            "currency": "change this currency to a foreign currency - Exs: JPY, AUD ",
            "source_uuids": ["uuid in the local body matches the one below"],
            "sources": [
                {
                    "uuid": "uuid in the local body",
                    "name": "Test AWS Source"
                }
            ]
        }
```
4. Delete the endpoint you are going to create based off your drafted body [DEL]http://127.0.0.1:8000/api/cost-management/v1/cost-models/UUID-to-delete/
5. Now ensure you can filter the cost models endpoint http://127.0.0.1:8000/api/cost-management/v1/cost-models/?currency=testing currency
6. Also test if anything not a currency receives the following repsonse
```
"errors": [
        {
            "detail": "test string" is not a valid choice.",
            "source": "currency",
            "status": 400
        }
    ]
```

---
Additional Context:
